specjbb/specjbb_run

```diff
@@ -258,24 +284,76 @@ concat_files()
 	done
 }
 
+reduce_data()
+{
+	temp_dir=$(mktemp -d /tmp/spec_reduce.XXXXXX)
+	files=`ls *_jvm_*_of_${1}`
+	index=1
+	prior=0
+	save_file=$temp_dir/worker
+	merge_file=$temp_dir/merge
+	results_file=specjbb_merged_jvms_${1}.csv
+	total_file=specjbb_total_jvms_${1}.csv
+	for results in $files; do
+		found=0
+		while IFS= read -r line
+		do
+			if [[ $line == *"TOTALS"* ]]; then
+					wh=`echo $line | awk '{print $5}'`
+					found=1	
+					continue
+				fi
+			if [ $found -eq 0 ]; then
+				continue;
+			fi
+			if [[ $line == *"throughput"* ]]; then
+				value=`echo $line | awk '{print $3}'`
+				echo ${wh} ${value} >> $save_file
+				found=0
+			fi
+		done < "$results"
+		if [ $index -gt 1 ]; then
+			join $results_file $save_file >> $merge_file
+			mv $merge_file $results_file
+			rm $save_file
+		else
+			mv $save_file $results_file
+		fi
+		let "index=$index+1"
+	done
+
+	while IFS= read -r data
+	do
+		wh_jvm=`echo $data | cut -d' ' -f 1`
+		values=`echo $data | cut -d' ' -f 2-`
+		total=0
+		for val in $values; do
+			total=`echo "$total + $val" | bc`
+		done
+		wh_jvm=`echo "$wh_jvm*$1" | bc`
+		echo $wh_jvm:$total >> $total_file
+	done < "$results_file"
+	rm -rf $temp_dir  $files
+}
+
 #
 # Execute specjbb.  Passed in is 
 #  syntax of $1
 #  ENDWH=<val>:STARTWH=<val> and so forth.
 #
 execute_specjbb()
 {
-	set_prop_file $1
+	set_prop_file $numb_jvms $1
 
 	#
 	# Start up all the jvms.
 	#	
 	for jvms in 0 `seq 1 1  $nr_jvms`; do
+		let "node_index=${jvms}+1"
```

Comment:
Why ${jvms}+1?  We should start on NUMA node 0 and increment from there for binding.